src/content-scope-features.js

```diff
@@ -5,25 +5,24 @@ import { PerformanceMonitor } from './performance'
 import injectedFeaturesCode from 'ddg:runtimeInjects'
 import platformFeatures from 'ddg:platformFeatures'
 
-function shouldRun () {
-    // don't inject into non-HTML documents (such as XML documents)
-    // but do inject into XHTML documents
-    // Should check HTMLDocument as Document is an alias for XMLDocument also.
-    if (document instanceof HTMLDocument === false && (
-        document instanceof XMLDocument === false ||
-        document.createElement('div') instanceof HTMLDivElement === false
-    )) {
-        return false
-    }
-    return true
-}
-
 let initArgs = null
 const updates = []
 const features = []
 const alwaysInitFeatures = new Set(['cookie'])
 const performanceMonitor = new PerformanceMonitor()
 
+// Check the document type in advance. That way, it minimises the risk of a
+// website breaking the `document instanceof ...` checks by changing
+// document.__proto__. In the future, it might be worth moving this to run even
+// earlier (and in the "isolated world" for the Chrome extension), to further
+// reduce that risk.
+const isHTMLDocument = (
```

Comment:
Can you leave in the comments about not wanting to inject into XML, that's super important to explain to newcomers why we don't run. You can reword it etc, just explain why we're doing the check at all.

- Maybe you've removed it as it's now just a variable that doesn't represent `shouldRun`?
   - If so I would consider renaming it that and making the code a function as you had it before?

I'd reduce this new comment wording a little tbh; I don't really think the 'in the future' bit is needed; if you really wish raise an issue for that.

Reply:
I've added the part of the comment back that you wanted and cut the word count down a bit.

Reply:
Can you just remove the last sentence and raise an issue? Nice to haves aren't really needed in the comments imo.
This check doesn't relate to 'privacy protections', can you reword somehow?

Reply:
>  This check doesn't relate to 'privacy protections', can you reword somehow?

Well, this variable is checked before calling load(), init() and update() for the content-scope-script features, and those features are primarily privacy protections. I don't mind rewording that again if you'd like, but would you like to make a suggestion?

> Can you just remove the last sentence and raise an issue?

I would prefer to keep that, I think it's useful. If in the future this issue comes up again, the comment will give us more of a clue.

Reply:
Done